src/Tribe/Main.php

```diff
@@ -6133,5 +6134,58 @@ public function register_plugin_autoload_paths( ) {
 
 			$this->get_autoloader_instance()->register_prefixes( $prefixes );
 		}
+
+		/**
+		 * Create a Filter Bar upsell tab.
+		 * 
+		 * @since TBD
+		 */
+		public function do_fbar_upsell_tab() {
+			// Bail if Filter Bar is already installed.
+			if ( class_exists( 'Tribe__Events__Filterbar__View' ) ) {
+				return;
+			}
+
+			tribe_assets(
+				self::instance(),
+				[
+					[
+						'tec-admin-filterbar-upsell',
+						'tec-admin-filterbar-upsell.css',
+					],
+				],
+				'admin_enqueue_scripts',
+			);
+
+			ob_start();
+			include_once $this->plugin_path . 'src/admin-views/filterbar/banners/filterbar-upsell.php';
+			$fbar_upsell_tab_html = ob_get_clean();
+
+			$fbar_upsell_tab = [
+				'info-box-description' => [
+					'type' => 'html',
+					'html' => $fbar_upsell_tab_html,
+				],
+			];
+			
+			/**
+			* Allows the fields displayed in the Filter Bar upsell tab to be modified.
+			*
+			* @since TBD
+			*
+			* @param array $fbar_upsell_tab Array of fields used to setup the Filter Bar upsell Tab.
+			*/
+			$fbar_upsell_fields = apply_filters( 'tribe_upgrade_fields', $fbar_upsell_tab );
```

Comment:
We need to namespace this correctly, `tec_events_` and have some more naming related to filterbar.

Reply:
namespace updated

Reply:
@EugeneKyale it wasn't the $variable that needed namespacing - it was the hook `tribe_upgrade_fields` -
which is also already in use [elsewhere](https://github.com/the-events-calendar/the-events-calendar/blob/master/src/Tribe/Main.php#L1361), so someone using it might break something unrelated.

Maybe `tec_events_filterbar_upgrade_content` would be more appropriate. Also, the use of the word "fields" here is misleading - there are no fields, just HTML content.

Reply:
Thanks @Camwyn I've namespaced the hook correctly.